### PR TITLE
Generic GestureRecognizer and TapGestureRecognizer hooks

### PR DESCRIPTION
**Pull request by Albert221**
See #190.

### CODE REVIEW

File: lib/src/gesture_recognizers.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
+part of 'hooks.dart';
+
+abstract class _GestureRecognizerHook<G extends GestureRecognizer>
```
Comment (Albert221) [inline lib/src/gesture_recognizers.dart:3]: Generic hook class is to make creating gesture detectors hooks involve as small code duplication as possible.

File: example/.gitignore [hunk 2]
```diff
@@ -23,6 +23,7 @@
 **/doc/api/
 .dart_tool/
 .flutter-plugins
+.flutter-plugins-dependencies
```
Comment (Albert221) [inline example/.gitignore:26]: I ignored this file as it should be ignored! :)

File: lib/src/hooks.dart [hunk 3]
```diff
@@ -9,10 +10,11 @@ import 'framework.dart';
 
 part 'animation.dart';
 part 'async.dart';
+part 'focus.dart';
```
Comment (Albert221) [inline lib/src/hooks.dart:13]: I sorted those `part` directives alphabetically.

File: test/gesture_detectors_test.dart [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_hooks/src/framework.dart';
+import 'package:flutter_hooks/src/hooks.dart';
+
+import 'mock.dart';
+
+void _testGestureRecognizer<T>({
```
Comment (Albert221) [inline test/gesture_detectors_test.dart:10]: Here is also one generic method that makes sure hook works as hooks should for all gesture detectors that would be tested here.

Comment (rrousselGit) [reply test/gesture_detectors_test.dart:10]: I don't like the idea of having such abstraction.
I'd prefer removing it and make such that the expectations don't depend on anything

File: test/gesture_detectors_test.dart [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  Type gestureRecognizer,
+  T Function() useHook,
+}) {
+  group('use$gestureRecognizer', () {
```
Comment (rrousselGit) [inline test/gesture_detectors_test.dart:14]: Could we add a test where the gesture detector is used, with a click & all?

File: test/gesture_detectors_test.dart (hunk 4 above)
Comment (Albert221) [reply test/gesture_detectors_test.dart]: Done!

File: test/gesture_detectors_test.dart (hunk 5 above)
Comment (Albert221) [reply test/gesture_detectors_test.dart]: Done!

**Comment by rrousselGit** [comment]
Closing as per https://github.com/rrousselGit/flutter_hooks/issues/190#issuecomment-703264898

I appreciate your work, but I just don't think that is the correct path for this kind of thing

**Pull request closed without merging**